conformance/utils/http/http.go

```diff
@@ -72,8 +73,29 @@ func MakeRequestAndExpectResponse(t *testing.T, r roundtripper.RoundTripper, gwA
 			req.Headers[name] = []string{value}
 		}
 	}
-	cReq, cRes, err := r.CaptureRoundTrip(req)
-	require.NoErrorf(t, err, "error making request")
+
+	var cReq *roundtripper.CapturedRequest
+	var cRes *roundtripper.CapturedResponse
+
+	// We expect eventual consistency once the resources are ready, so wait until
+	// we've made a successful request and gotten a response with desired status
+	// before making assertions on the response body.
+	require.Eventually(t, func() bool {
```

Comment:
Do we want to test eventually **consistent**? Technically a proxy doing a 50/50 200 or 404 passes currently. We could do a check that does backoff on failures, and on success retry immediately and ensure we get the "success" N times.

In Istio tests, we do all of ours this way fwiw and it seems to work pretty well. We do exponential backoff and ensure we get 3 successes consecutively

Reply:
@howardjohn if you would, please take a look at latest commits and let me know what ya think. Each test that needs to make assertions using this path now looks like this:

<img width="500" alt="image" src="https://user-images.githubusercontent.com/3476400/160709210-d765b71f-edbb-4bea-bab5-e1214149330f.png">